packages/wonder-blocks-layout/components/media-layout.js

```diff
@@ -52,104 +67,67 @@ type State = {|
 // fall back to this state.
 const DEFAULT_SIZE = "large";
 
+type CombinedProps = {|...Props, ...Context|};
+
 /**
  * `MediaLayout` is responsible for changing the rendering of contents at
  * differently sized viewports.  `MediaLayoutContext.Provider` can be used
  * to specify different breakpoint configurations.  By default it uses
  * `MEDIA_DEFAULT_SPEC`.  See media-layout-context.js for additiional options.
  */
-export default class MediaLayout extends React.Component<Props, State> {
-    watchHandlers: ?{
-        [query: string]: any,
-    };
-
-    static WATCHERS: {
-        [query: string]: any,
-    } = {};
-
-    state = {
-        size: undefined,
-    };
+class MediaLayoutInternal extends React.Component<CombinedProps, State> {
+    cleanupThunks: Array<() => void>;
+
+    constructor(props: CombinedProps) {
+        super(props);
+        this.state = {
+            size: undefined,
+        };
+        this.cleanupThunks = [];
+    }
 
-    componentWillUnmount() {
-        // We go through the component and remove all of the listeners
-        // that getCurrentSize attached.
-        for (const query of Object.keys(MediaLayout.WATCHERS)) {
-            const watcher = MediaLayout.WATCHERS[query];
-
-            if (watcher && this.watchHandlers) {
-                const {watchHandlers} = this;
-                const handler = this.watchHandlers[query];
-                watcher.removeListener(handler);
-                delete watchHandlers[query];
+    componentDidMount() {
+        // TODO(WB-534): handle changes to mediaSpec prop
+        for (const [size, spec] of ((Object.entries(
+            this.props.mediaSpec,
+        ): any): Array<[MediaSize, {query: string}]>)) {
```

Comment:
Could you put the result of `Object.entries` into a const before this loop, just to make this more readable. Like:

`const specs: Array<[MediaSize, {query: string}]> = Object.entries(this.props.mediaSpec);`

..should work, though you "might" have to cast the rhs to `any` as you do now.